presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetCompressionUtils.java

```diff
@@ -86,30 +89,25 @@ private static Slice decompressGzip(Slice input, int uncompressedSize)
         }
     }
 
+    // decompress LZO-compressed data using HADOOP-LZO codec
     private static Slice decompressLZO(Slice input, int uncompressedSize)
+            throws IOException
     {
-        LzoDecompressor lzoDecompressor = new LzoDecompressor();
-        long totalDecompressedCount = 0;
-        // over allocate buffer which makes decompression easier
-        byte[] output = new byte[uncompressedSize + SIZE_OF_LONG];
-        int outputOffset = 0;
-        int inputOffset = 0;
-        int cumulativeUncompressedBlockLength = 0;
-
-        while (totalDecompressedCount < uncompressedSize) {
-            if (totalDecompressedCount == cumulativeUncompressedBlockLength) {
-                cumulativeUncompressedBlockLength += Integer.reverseBytes(input.getInt(inputOffset));
-                inputOffset += SIZE_OF_INT;
-            }
-            int compressedChunkLength = Integer.reverseBytes(input.getInt(inputOffset));
-            inputOffset += SIZE_OF_INT;
-            int decompressionSize = decompress(lzoDecompressor, input, inputOffset, compressedChunkLength, output, outputOffset);
-            totalDecompressedCount += decompressionSize;
-            outputOffset += decompressionSize;
-            inputOffset += compressedChunkLength;
-        }
-        checkArgument(outputOffset == uncompressedSize);
-        return wrappedBuffer(output, 0, uncompressedSize);
+        final BytesInput decompressed;
+        LzoCodec codec = new LzoCodec();
+        codec.setConf(new Configuration());
+        org.apache.hadoop.io.compress.Decompressor decompressor = codec.createDecompressor();
+
+        // input data
+        byte[] byteArray = (byte[]) input.getBase();
+        int inputOffset = (int) input.getAddress() - ARRAY_BYTE_BASE_OFFSET;
+        ByteArrayBytesInput bytes = new ByteArrayBytesInput(byteArray, inputOffset, input.length());
```

Comment:
`bytes` is only used to be converted to InputStream, use `InputStream in = new ByteArrayInputStream(byteArray, inputOffset, input.length())`